packages/compass-components/src/components/item-action-controls.tsx

```diff
@@ -47,11 +48,20 @@ const iconContainerStyle = css({
 
 // Using important here because leafygreen / emotion applies styles in the order
 // that doesn't allow our styles override theirs
-const iconButtonSmallStyle = css({
-  flex: 'none',
-  width: `${spacing[4]}px !important`,
-  height: `${spacing[4]}px !important`,
-});
+const buttonSizeStyle: Record<ItemActionButtonSize, string | undefined> = {
+  default: undefined,
+  small: css({
+    flex: 'none',
+    width: `${spacing[4]}px !important`,
+    height: `${spacing[4]}px !important`,
+  }),
+  xsmall: css({
```

Comment:
add xsmall icon buttons, should be moved after https://jira.mongodb.org/browse/COMPASS-6306